parser/unparser.go

```diff
@@ -444,3 +471,76 @@ func bytesToOctets(byteVal []byte) string {
 	}
 	return b.String()
 }
+
+// Takes in a function then inserts a newline based on the rules provided in formatting options
+func (un *unparser) wrapForOperator(fun string) bool {
+	_, wrapOperatorExists := un.options.operatorsToWrapOn[fun]
+	lineLength := un.str.Len() - un.lastWrappedIndex
+	if wrapOperatorExists && lineLength >= un.options.wrapColumn {
+		un.lastWrappedIndex = un.str.Len()
+		un.str.WriteString("\n")
+		return true
+	}
+	return false
+}
+
+// Defined defaults for the formatting options
+var (
+	defaultWrapColumn = 80
+)
+
+// UnparserFormattingOption is a funcitonal option for configuring the output formatting
+// of the Unparse function.
+type UnparserFormattingOption func(*formattingOption) (*formattingOption, error)
+
+// Internal representation of the UnparserFormattingOption type
+type formattingOption struct {
+	wrapColumn        int // Defaults to 80
+	operatorsToWrapOn map[string]bool
+}
+
+// WrapOnColumn returns an UnparserFormattingOption with the column limit set.
+// Word wrapping will be performed when a line's length exceeds the limit
+// for operators set by WrapOnOperators function.
+//
+// Example usage:
+//
+//	Unparse(expr, sourceInfo, WrapColumn(3), WrapOnOperators(Operators.LogicalAnd))
```

Comment:
Yeah 3 is the number of chars. Currently, WrapColumn happens after the operator but I think it'd be best to make it configurable. Thinking about this, it would make sense to support wrapping at a blank space since that's a pretty common use case.

Reply:
In this case, I think I'm looking for an updated example. As a refinement, you could also support wrap at space as an option as well.